go/vt/vttablet/tabletmanager/vreplication/vreplicator.go

```diff
@@ -350,8 +350,18 @@ func (vr *vreplicator) buildColInfoMap(ctx context.Context) (map[string][]*Colum
 	return colInfoMap, nil
 }
 
-func (vr *vreplicator) readSettings(ctx context.Context) (settings binlogplayer.VRSettings, numTablesToCopy int64, err error) {
-	settings, err = binlogplayer.ReadVRSettings(vr.dbClient, vr.id)
+// Same as readSettings, but stores some of the results on this vr.
+func (vr *vreplicator) loadSettings(ctx context.Context, dbClient *vdbClient) (settings binlogplayer.VRSettings, numTablesToCopy int64, err error) {
```

Comment:
I don't see the need for these two function changes, what am I missing?

Reply:
I want to be able to re-use logic in `vreplicator.go` for setting things like `sql_mode` and disabling FK checks. I want to use that logic in each new parallel MySQL conn that's created in `vcopier.go`. The current `vreplicator.go` code makes that a little difficult because it changes state in `vreplicator`. 

To address that I move the logic I wanted to re-use into a stateless fn (`readSettings`), and move the stateful stuff into `loadSettings`.